core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/TerminationTriggerService.java

```diff
@@ -90,12 +89,7 @@ private void handleIfFlowLogExistsForTermination(Stack stack, boolean forced, Fl
     private boolean isRunningFlowForced(FlowLog fl) {
         ClassValue payloadType = fl.getPayloadType();
         if (payloadType != null && payloadType.isOnClassPath() && TerminationEvent.class.equals(payloadType.getClassValue())) {
-            TerminationEvent payload;
-            if (fl.getPayloadJackson() != null) {
-                payload = JsonUtil.readValueWithJsonIoFallback(fl.getPayloadJackson(), fl.getPayload(), TerminationEvent.class);
-            } else {
-                payload = (TerminationEvent) JsonReader.jsonToJava(fl.getPayload());
-            }
+            TerminationEvent payload = JsonUtil.readValueUnchecked(fl.getPayloadJackson(), TerminationEvent.class);
```

Comment:
Sanity check: What if `fl.getPayloadJackson() == null` (if such a case is possible at all)? Can Jackson `ObjectMapper` tolerate it, or will rather result in an NPE?

Reply:
ObjectMapper throws an `IllegalArgumentException:  argument "content" is null`

Reply:
And everything is possible ;)

Reply:
If the serialization is not successful then it can happen. By the way, it was the case earlier with Cedarsoft's JsonWriter falling back to ObjectMapper.

Reply:
And line 93 will throw an NPE (but it was the case with JsonReader as well)